.travis.yml

```diff
@@ -34,7 +34,7 @@ install:
 before_script:
   - set -o errexit; source .travis/test_05_before_script.sh
 script:
-  - set -o errexit; source .travis/test_06_script.sh
+  - if [ $SECONDS -gt 1200 ]; then set +o errexit; echo "Travis early exit to cache current state"; false; else set -o errexit; source .travis/test_06_script.sh; fi
```

Comment:
Should this be in the `before_script`?

Reply:
`before_script` does the depends build, which should be cached, so no?

Reply:
I didn't know chache was discared during a premature exit. If so, then this makes sense.